pkg/pods/hoist_launchable.go

```diff
@@ -335,7 +335,18 @@ func (hoistLaunchable *HoistLaunchable) extractTarGz(fp *os.File, dest string) (
 		}
 		fpath := path.Join(dest, hdr.Name)
 
-		if hdr.FileInfo().IsDir() {
+		switch hdr.Typeflag {
+		case tar.TypeSymlink, tar.TypeLink:
+			// hard links are intentionally treated as symlinks, because the
+			// target of a hardlink might be later in the tarball and not
+			// unpacked yet
+			// creating a hardlink in that situation would raise an error
+			// because the hardlink target doesn't exist
+			err = os.Symlink(hdr.Linkname, fpath)
+			if err != nil {
+				return util.Errorf("Unable to create destination symlink %s (to %s) when unpacking %s: %s", fpath, hdr.Linkname, fp.Name(), err)
+			}
+		case tar.TypeDir:
```

Comment:
We may want to extract this as a standalone utility at some point. This function is really getting gnarly.